File: core-java/src/main/java/hello/core/member/MemoryMemberRepository.java
package hello.core.member;

import java.util.HashMap;
import java.util.Map;

public class MemoryMemberRepository implements MemberRepository{

    // 실무에서는  ConcurrentHashMap 사용하여 동시성 이슈를 해결해야 함.
    private static final Map<Long, Member> store = new HashMap<>();
    @Override
    public void save(Member member) {
        store.put(member.getId(), member);
    }

    @Override
    public Member findById(Long memberId) {
        return store.get(memberId);
    }
}

File: core-kotlin/src/main/kotlin/hello/core/member/MemoryMemberRepository.kt
package hello.core.member

import org.springframework.stereotype.Component

@Component
class MemoryMemberRepository() : MemberRepository {
    companion object {
        val store = hashMapOf<Long, Member>()
    }
    override fun save(member: Member) {
        store[member.id] = member
    }

    override fun findById(memberId: Long): Member? {
        return store[memberId]
    }
}